pkg/gatewayserver/gatewayserver.go

```diff
@@ -604,7 +604,15 @@ func (gs *GatewayServer) startDisconnectOnChangeTask(conn connectionEntry) {
 				},
 			})
 			if err != nil {
-				log.FromContext(ctx).WithError(err).Warn("Failed to get gateway")
+				if errors.IsUnauthenticated(err) || errors.IsPermissionDenied(err) {
+					// Since there is an active connection, the `Get` request will not return a 404 as the gateway existed during the connect. Instead,
```

Comment:
Why would it not when the gateway gets deleted while it is connected?

Nit; 404 is HTTP which isn't applicable here

Reply:
The IS fails on the rights assertion before it checks if the entity exists. Updated the comment.